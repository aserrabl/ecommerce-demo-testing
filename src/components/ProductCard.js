import React from 'react';
import { Card, CardContent, CardMedia, Typography, Button } from '@mui/material';

const ProductCard = ({ product, onAddToCart }) => {

  /** 
  if (process.env.REACT_APP_PW_TEST){
    product.image = null
  }
  */
  return (
    <Card sx={{ maxWidth: 345, m: 2 }}>
      <CardMedia
        component="img"
        height="140"
        image={product.image}
        alt={product.name}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {product.name}
        </Typography>
        <Typography variant="body2" color="text.secondary">
          ${product.price.toFixed(2)}
        </Typography>
        <Button 
          variant="contained" 
          color="primary" 
          onClick={() => onAddToCart(product)}
          sx={{ mt: 2 }}
          data-testid={`add-to-cart-${product.id}`}
        >
          Add to Cart
        </Button>
      </CardContent>
    </Card>
  );
};

export default ProductCard;
