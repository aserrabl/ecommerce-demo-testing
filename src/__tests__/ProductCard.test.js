import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ProductCard from '../components/ProductCard';

describe('Snapshot - ProductCard ', () => {
  const mockProduct = {
    id: 1,
    name: 'Test Product',
    price: 99.99,
    image: 'test-image.jpg'
  };

  it('should match snapshot', () => {
    const { container } = render(
      <ProductCard 
        product={mockProduct}
        onAddToCart={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with different product data', () => {
    const differentProduct = {
      id: 2,
      name: 'Another Product',
      price: 149.99,
      image: 'another-image.jpg'
    };

    const { container } = render(
      <ProductCard 
        product={differentProduct}
        onAddToCart={() => {}}
      />
    );
    
    expect(container).toMatchSnapshot();
  });
});

describe('Unit - ProductCard', () => {
  const mockProduct = {
    id: 1,
    name: 'Test Product',
    price: 99.99,
    image: 'test-image.jpg'
  };

  it('calls function to product to cart when add card button is clicked', () => {
    const mockOnAddToCart = jest.fn();
    render(
      <ProductCard
        product={mockProduct}
        onAddToCart={mockOnAddToCart}
      />
    );

    const addToCartButton = screen.getByRole('button', { name: /add to cart/i });
    fireEvent.click(addToCartButton);

    expect(mockOnAddToCart).toHaveBeenCalled();
  });
});