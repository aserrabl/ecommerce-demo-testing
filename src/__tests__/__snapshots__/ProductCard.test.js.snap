// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot - ProductCard  should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-iymoya-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Test Product"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="test-image.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Test Product
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        $
        99.99
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-lll1vm-MuiButtonBase-root-MuiButton-root"
        data-testid="add-to-cart-1"
        tabindex="0"
        type="button"
      >
        Add to Cart
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Snapshot - ProductCard  should match snapshot with different product data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-iymoya-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="Another Product"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="140"
      src="another-image.jpg"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Another Product
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        $
        149.99
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-lll1vm-MuiButtonBase-root-MuiButton-root"
        data-testid="add-to-cart-2"
        tabindex="0"
        type="button"
      >
        Add to Cart
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
